actionpack/lib/action_dispatch/http/response.rb

```diff
@@ -448,7 +448,7 @@ def parsed_content_type_header
     end
 
     def set_content_type(content_type, charset)
-      type = (content_type || "").dup
+      type = +(content_type || "")
```

Comment:
Note that actually `String#+@` is not the same with `String#dup` if a string is mutable.

In this case, `content_type` is always passed `ContentTypeHeader#media_type` which is mutable, so if changed to `+(content_type || "")`, it will mutates `ContentTypeHeader#media_type` directly, but it is also one-time persed_content_type_header instance, so direct mutating is safe.

So how about only unmutating `+""`?

```suggestion
      type = content_type || +""
```

Reply:
Oh, I didn't know it was guaranteed to have `content_type` mutable. If that's the case, indeed it's better to only unfreeze the empty string.

Reply:
Based on the failing tests, it looks like they depend on making `content_type` mutable. The failures seem legitimate. Should we revert the last commit?

Reply:
hm... looks like `Mime[:html].to_s` is not always mutable.

https://github.com/rails/rails/blob/2db687ff87abae33086744d9ac97322856d34fc2/actionpack/lib/action_dispatch/http/response.rb#L487-L489